engine/source/private/core/engine.cpp

```diff
@@ -10,6 +12,12 @@ namespace lumina
 {
     void Engine::Initialize()
     {
+        fileIO = std::make_unique<lumina::FileIO>();
+        if (!fileIO)
+        {
+            throw std::runtime_error("FileIO is not initialized, this should never happen!");
```

Comment:
I assume we eventually want to catch all exceptions so we can display them to the user and not have the engine CTD

Reply:
The engine should not crash when these "systems" are not available. But for now I think just throwing is fine, to get started.